aws/data_source_aws_ecr_authorization_token_test.go

```diff
@@ -0,0 +1,60 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccAWSEcrDataSource_ecrAuthorizationToken(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "data.aws_ecr_authorization_token.repo"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAwsEcrAuthorizationTokenDataSourceBasicConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "authorization_token"),
+					resource.TestCheckResourceAttrSet(resourceName, "proxy_endpoint"),
+					resource.TestCheckResourceAttrSet(resourceName, "expires_at"),
+					resource.TestCheckResourceAttrSet(resourceName, "user_name"),
+					resource.TestMatchResourceAttr(resourceName, "user_name", regexp.MustCompile(`AWS`)),
+					resource.TestCheckResourceAttrSet(resourceName, "password"),
+				),
+			},
+			{
```

Comment:
I think the idea of having two separate test is to check whether the authorization token was successfully obtained in both cases (with or without a registry_id paramter). If we leave only one test I think we will not be able to check whether the token was obtained in the other case. Personally, I think it makes sense to keep it but I understand that it can quite repetitive. If you still prefer to remove please let me know and I'll be happy to remove it. Thanks!

Reply:
ah yes you're right, I overlooked there's a different config being referenced in each step. makes sense to have both. my only suggestion then is regarding the first check made at L34, we can use the function in my above comment to be able to verify the data_source against the resource

Reply:
fixed in 182dfdb55fe1e3648e7313c15a2509a765eec6c0